Lib/test/test_inspect.py

```diff
@@ -4148,6 +4148,16 @@ def func(*args, **kwargs):
         sig = inspect.signature(func)
         self.assertEqual(str(sig), '(self, a, b=1, /, *args, c, d=2, **kwargs)')
 
+    def test_base_class_have_text_signature(self):
+        # see issue 43118
+        from io import BufferedReader
+        class MyBufferedReader(BufferedReader):
+            """buffer reader class."""
+
+        self.assertTrue(hasattr(BufferedReader, '__text_signature__'))
+        sig = inspect.signature(MyBufferedReader)
+        self.assertIsNotNone(sig)
```

Comment:
This is a rather low standard. Perhaps we can test that the signature is the same as for io.BufferedReader, and test that the DEFAULT_BUFFER_SIZE constant appears in the signature correctly? That way, we'll notice if io is changed to no longer reference the constant, which would cause this test to not test what it needs to test.